pkg/controller/servingcert/starter/starter.go

```diff
@@ -50,33 +53,36 @@ func StartServiceServingCertSigner(ctx context.Context, controllerContext *contr
 	if err != nil {
 		return fmt.Errorf("failed to create config client: %w", err)
 	}
-
 	configInformers := configexternalinformers.NewSharedInformerFactory(configClient, 10*time.Minute)
-
 	stopChan := ctx.Done()
-	klog.Infof("ShortCertRotation: fetching FeatureGates")
-	featureGateAccessor := featuregates.NewFeatureGateAccess(
-		status.VersionForOperandFromEnv(), "0.0.1-snapshot",
-		configInformers.Config().V1().ClusterVersions(), configInformers.Config().V1().FeatureGates(),
-		controllerContext.EventRecorder,
-	)
-	go featureGateAccessor.Run(ctx)
-	configInformers.Start(stopChan)
-
-	var featureGates featuregates.FeatureGate
-	select {
-	case <-featureGateAccessor.InitialFeatureGatesObserved():
-		featureGates, _ = featureGateAccessor.CurrentFeatureGates()
-	case <-time.After(1 * time.Minute):
-		klog.Errorf("timed out waiting for FeatureGate detection")
-		return fmt.Errorf("timed out waiting for FeatureGate detection")
-	}
 
 	minTimeLeftForCert := time.Hour
 	certificateLifetime := 2 * 365 * 24 * time.Hour
-	if featureGates.Enabled(features.FeatureShortCertRotation) {
-		minTimeLeftForCert = time.Hour
-		certificateLifetime = time.Hour * 2
+	if ok, err := IsMicroshift(ctx, kubeClient); err == nil && !ok {
```

Comment:
Can we avoid creating a precedent for adding MicroShift special cases to OCP components?

The cluster-config-operator includes code to render configv1 FeatureGate manifest files. Could we instead add an option to service-ca-operator to provide a FeatureGate manifest to be used instead of loading them from the API? MicroShift would be able to configure that option appropriately, and service-ca-operator would not need to detect or even be aware of MicroShift.

Reply:
@pmtk wdyt?

Reply:
These are the components run by Microshift, among them `service-ca` (the operand, not the operator)

https://github.com/openshift/microshift/tree/main/assets/components

Afaict, the only other controller that Microshift runs that is feature gate aware is the `csi-snapshot-controller`, which takes gates as a flag
https://github.com/openshift/csi-external-snapshotter/blob/ac82cafc95b301f67f46ee0db93720d55177a19b/cmd/snapshot-controller/main.go#L154

Microshift just doesn't set the flag.

Looking back at https://github.com/openshift/service-ca-operator/pull/245, we should have added the FeatureGate informer to the operator (like we did), but added a `feature-gates` flag to the controller rather than have it directly observe the FeatureGate config.

Reply:
I think that makes sense, it would be more elegant solution